lib/backend/k8s/syncer.go

```diff
@@ -437,6 +477,22 @@ func (syn *kubeSyncer) parseNamespaceEvent(e watch.Event) []model.KVPair {
 	return []model.KVPair{*rules, *tags, *labels}
 }
 
+func (syn *kubeSyncer) parseNodeEvent(e watch.Event) *model.KVPair {
+	node, ok := e.Object.(*k8sapi.Node)
+	if !ok {
+		log.Panicf("Invalid node event. Type: %s, Object: %+v", e.Type, e.Object)
+	}
+
+	kvp := syn.kc.converter.k8sNodeToCalico(node)
+
+	switch e.Type {
+	default:
+		syn.labelCache[kvp.Key.(model.NodeKey).Hostname] = kvp.Value.(*model.Node).Labels
```

Comment:
Hmm, a switch with only a default seems a bit weird!

We need to handle deletes by setting the `kvp.Value` to nil.

Also, the `labelCache` field is meant for Pods only - we don't need to do the equivalent for nodes since we care about all updates to the Node object.

`parseNetworkPolicyEvent` is probably the closest to what you're looking for.